htdocs/projet/tasks/time.php

```diff
@@ -1566,13 +1566,22 @@ function setDetailVisibility() {
 		$varpage = empty($contextpage) ? $_SERVER["PHP_SELF"] : $contextpage;
 		$selectedfields = $form->multiSelectArrayWithCheckbox('selectedfields', $arrayfields, $varpage, getDolGlobalString('MAIN_CHECKBOX_LEFT_COLUMN')); // This also change content of $arrayfields
 
+		// Check if optional invoiceable exists for the project tasks
+		$object->fetch_optionals();
+		$invoiceable = false;
+		if ( isset($object->array_options) && array_key_exists('options_invoiceable', $object->array_options)) {
+			$invoiceable = true;
+		}
 		$sql = "SELECT t.rowid, t.fk_element, t.element_date, t.element_datehour, t.element_date_withhour, t.element_duration, t.fk_user, t.note, t.thm,";
 		$sql .= " t.fk_product,";
 		$sql .= " pt.ref, pt.label, pt.fk_projet,";
 		$sql .= " u.lastname, u.firstname, u.login, u.photo, u.gender, u.statut as user_status,";
 		$sql .= " il.fk_facture as invoice_id, inv.fk_statut,";
 		$sql .= " p.fk_soc,s.name_alias,";
-		$sql .= " t.invoice_line_id";
+		$sql .= " t.invoice_line_id,";
+		if ($invoiceable) {
+			$sql .= " efpt.invoiceable as options_invoiceable";
```

Comment:
This field does not exist by default in dolibarr so should not be used in default code.